common/src/main/java/org/ergoplatform/appkit/Bip32Serialization.java

```diff
@@ -0,0 +1,95 @@
+package org.ergoplatform.appkit;
+
+import org.bouncycastle.jcajce.provider.digest.RIPEMD160;
+import org.ergoplatform.appkit.NetworkType;
+import org.ergoplatform.wallet.Constants;
+import org.ergoplatform.wallet.secrets.DerivationPath;
+import org.ergoplatform.wallet.secrets.ExtendedPublicKey;
+import org.ergoplatform.wallet.secrets.ExtendedSecretKey;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Arrays;
+
+import scala.collection.immutable.Seq;
+import scorex.crypto.hash.Sha256;
+import scorex.util.encode.Base16$;
+
+/**
+ * Methods to serialize and deserialize key according to bip32 standard
+ * https://en.bitcoin.it/wiki/BIP_0032
+ */
+public class Bip32Serialization {
```

Comment:
Why this is not suitable?
https://github.com/ergoplatform/ergo/blob/93f34e3b9ee6e94505c60d6a5a900ca9d102db86/ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/ExtendedSecretKey.scala#L90

Reply:
It is a viable serialization/deserialization approach, but not compatible to bip-32 and that's how Yoroi, Nautilus and Paper Waller implemented it. So in order to be compatible to the already existing wallets, we need bip-32 compatible serialization and hex encoding.